applications/luci-app-ddns/luasrc/view/ddns/system_status.htm

```diff
@@ -95,56 +89,62 @@
 				network = '<em><%:Config error%></em>';
 			else
 				network = data[j].iface;
-
-			tbl.appendChild(E('<div class="tr cbi-section-table-row cbi-rowstyle-%d">'.format(((j + i) % 2) + 1), [
-					E('<div class="td">', [ E('<strong>', configuration) ]),
-					E('<div class="td">', next_update),
-					E('<div class="td">', lookup),
-					E('<div class="td">', registered_ip),
-					E('<div class="td">', network)
-					]));
-		}
-
-		if (tbl.firstElementChild === tbl.lastElementChild || (data[0].enabled == 0 && tbl.childNodes.length == 2) ) {
-			var br = '<br />';
-			var msg = '<%:There is no service configured.%>';
-			if (tbl.firstElementChild !== tbl.lastElementChild)
-				br = '';
-			msg = br + msg;
-			tbl.appendChild(E('<div class="tr cbi-section-table-row">', [
-					E('<div class="td">', [ E('<em>', msg) ])
-					]));
+			
+			rows.push([
+					'<strong>' + configuration + '</strong>',
+					next_update,
+					lookup,
+					registered_ip,
+					network
+					]);
+					
+			if (rows.length > 1) {
+				var legendtb = document.getElementById('ddns_status_table_legend');
+				legendtb.setAttribute("style", "display:table-row");
+			}
+			
+			cbi_update_table(tbl, rows, '<%:There is no service configured.%>');
 		}
+		
 	}
-
-	// force to immediate show status (not waiting for XHR.poll)
-	XHR.get('<%=url([[admin]], [[services]], [[ddns]], [[status]])%>', null,
-		function(x, data) {
-			if (data) { _data2elements(x, data); }
-		}
-	);
-
-	XHR.poll(15, '<%=url([[admin]], [[services]], [[ddns]], [[status]])%>', null,
-		function(x, data) {
-			if (data) { _data2elements(x, data); }
-		}
-	);
+	
+	var mutationObserver = new MutationObserver(function(mutations) {
+		// force to immediate show status (not waiting for XHR.poll)
+		XHR.get('<%=url([[admin]], [[services]], [[ddns]], [[status]])%>', null,
+			function(x, data) {
+				if (data) { _data2elements(x, data); }
+			}
+		);
+		
+		//start polling data every 30 second, this doesn't change so much
+		XHR.poll(30, '<%=url([[admin]], [[services]], [[ddns]], [[status]])%>', null,
+			function(x, data) {
+				if (data) { _data2elements(x, data); }
+			}
+		);
+		this.disconnect()
+	});
+	
+	mutationObserver.observe(document.getElementById("wifi_assoclist_table"), {
+		childList: true,
+	});
```

Comment:
Please remove the `MutationObserver` stuff, the XHR stuff has been fixed in a generic way now.

Reply:
so i can keep the xhr get ?

Reply:
No, just the sole XHR.poll(), no extraneous get etc. required.